File: python/leetcode/268. Missing Number.py
class Solution:
    def missingNumber(self, nums: List[int]) -> int:
        max = nums[0]
        map = set()
        
        for i in range(len(nums)):
            if(nums[i] > max):
                max = nums[i]
            map.add(nums[i])
            
        for i in range(max + 1):
            if i not in map:
                return i
        
        return max + 1
        

File: c++/leetcode/268. Missing Number.cpp
class Solution
{
public:
  int missingNumber(vector<int> &nums)
  {
    int max = nums[0];
    set<int> st;
    int pos = 0;

    while (pos != nums.size())
    {
      if (nums[pos] > max)
        max = nums[pos];
      st.insert(nums[pos]);
      pos++;
    }

    for (int i = 0; i <= max; i++)
    {
      if (!st.count(i))
      {
        return i;
      }
    }

    return max + 1;
  }
};